core/data/clients/InfluxEvent.go

```diff
@@ -0,0 +1,169 @@
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * @microservice: core-data-go library
+ * @author: Ryan Comer, Dell
+ * @version: 0.5.0
+ *******************************************************************************/
+package clients
+
+import (
+	"fmt"
+	"strconv"
+	"github.com/edgexfoundry/edgex-go/core/domain/models"
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+	"github.com/influxdata/influxdb/client/v2"
+)
+
+// Struct that wraps an event to handle DBRefs
+type InfluxEvent struct {
+	models.Event
+}
+
+// Custom marshaling into influxdb
+func (ie InfluxEvent) Get() (interface{}, error) {
+	// Turn the readings into DBRef objects
+	var readings []mgo.DBRef
+	for _, reading := range ie.Readings {
+		readings = append(readings, mgo.DBRef{Collection: READINGS_COLLECTION, Id: reading.Id})
+	}
+
+	return struct {
+		ID       bson.ObjectId `json:"_id,omitempty"`
+		Pushed   int64         `json:"pushed"`
+		Device   string        `json:"device"` // Device identifier (name or id)
+		Created  int64         `json:"created"`
+		Modified int64         `json:"modified"`
+		Origin   int64         `json:"origin"`
+		Schedule string        `json:"schedule,omitempty"` // Schedule identifier
+		Event    string        `json:"event"`              // Schedule event identifier
+		Readings []mgo.DBRef   `json:"readings"`           // List of readings
+	}{
+		ID:       ie.ID,
+		Pushed:   ie.Pushed,
+		Device:   ie.Device,
+		Created:  ie.Created,
+		Modified: ie.Modified,
+		Origin:   ie.Origin,
+		Schedule: ie.Schedule,
+		Event:    ie.Event.Event,
+		Readings: readings,
+	}, nil
+}
+
+// Custom unmarshaling out of influxdb
+func (ie *InfluxEvent) Set(raw bson.Raw) error {
+	decoded := new(struct {
+		ID       bson.ObjectId `json:"_id,omitempty"`
+		Pushed   int64         `json:"pushed"`
+		Device   string        `json:"device"` // Device identifier (name or id)
+		Created  int64         `json:"created"`
+		Modified int64         `json:"modified"`
+		Origin   int64         `json:"origin"`
+		Schedule string        `json:"schedule,omitempty"` // Schedule identifier
+		Event    string        `json:"event"`              // Schedule event identifier
+		Readings []mgo.DBRef   `json:"readings"`           // List of readings
+	})
+
+	bsonErr := raw.Unmarshal(decoded)
+	if bsonErr != nil {
+		return bsonErr
+	}
+
+	// Copy over the non-DBRef fields
+	ie.ID = decoded.ID
+	ie.Pushed = decoded.Pushed
+	ie.Device = decoded.Device
+	ie.Created = decoded.Created
+	ie.Modified = decoded.Modified
+	ie.Origin = decoded.Origin
+	ie.Schedule = decoded.Schedule
+	ie.Event.Event = decoded.Event
+
+	// De-reference the DBRef fields
+	ic, err := getCurrentInfluxClient()
+	if err != nil {
+		fmt.Println("Error getting a influxdb client: " + err.Error())
```

Comment:
No prits on stdout - please use logs

Reply:
Ok.
Should I use [log](https://golang.org/pkg/log/) library? 
or [zap](https://github.com/uber-go/zap)?

Reply:
Zap would be preferable. @feclare can point you to EdgeX logging package, if one exists (I am not sure)

Reply:
For code consistency I should use support/logging-client, as it is what core-data uses. But in this case you are only using two log messages, one that can be removed as is printing the response (line 160) and another before return an error. The error doesn't need the log either as the caller will manage the error and log as needed.

Reply:
@feclare 
Ok, I make sense. thank's